src/calculateCoordinates.ts

```diff
@@ -0,0 +1,58 @@
+import NodeGeocoder from 'node-geocoder';
+import * as dotenv from 'dotenv';
+// import { log } from '.';
+import { ERROR_MESSAGES, INVALID_API_KEY_ERROR } from './conf/constants';
+dotenv.config();
+
+export const getCoordinates = async (
+  address: string,
+  apiKey: string
+): Promise<NodeGeocoder.Entry[]> => {
+  try {
+    const options: NodeGeocoder.Options = {
+      provider: 'google',
+      // Optional depending on the providers
+      httpAdapter: 'https', // Default
+      apiKey, // for Mapquest, OpenCage, Google Premier
+      formatter: null, // 'gpx', 'string', ...
+    };
+    const geocoder = NodeGeocoder(options);
+    const fetchedAddr = await geocoder.geocode(address);
+    if (fetchedAddr.length === 0) {
+      throw new Error(ERROR_MESSAGES.INVALID_ADDRESS);
+    }
+    if (fetchedAddr.length > 1) {
+      throw new Error(ERROR_MESSAGES.INCOMPLETE_ADDRESS);
+    }
+    return fetchedAddr;
+  } catch (err) {
+    if (
+      err instanceof Error &&
+      err?.toString().includes(INVALID_API_KEY_ERROR)
+    ) {
+      throw new Error(ERROR_MESSAGES.INVALID_API_KEY);
+    }
+    throw err;
```

Comment:
Looks like we are bubbling unknown Geocoder errors to consumers on the API response. Is this desirable? Why?

Reply:
Good question, since i am already converting some known errors for API consumers, i thought converting other errors (non `constants.ts` file errors) to generic one, i might lose some useful information.
I honestly didn't encounter other errors, but it can be converted to generic error rather than showing entire stack trace to the API consumers,.

Reply:
Some known errors, i am able to convert them for API consumers, all other errors I just forwarded (i thought I might lose some information from the full error), but it can be converted to generic one - this way we can avoid sending full stack trace to the consumer

Reply:
> but it can be converted to generic one - this way we can avoid sending full stack trace to the consumer
This is what I was looking for. We can log the Geocode error inside the `getCoordinates` method and mask it with our own error definition, in that way we create a solid abstraction layer

Reply:
Sure, will give a try, stay tuned!

Reply:
Pushed a change for using generic error